### ISSUE TITLE
select2_json_from_api behaviour when form is reloaded 

### ISSUE THREAD

**Issue by miquelangeld** [Laravel-Backpack/CRUD#5557]
Hi, I'm  using select2_json_from_api, here is my code

```
$this->crud->addField([
            'label' => 'Booking', // Table column heading
            'type' => 'select2_json_from_api',
            'name' => 'booking', 
            'entity' => false, 
            'attribute' => 'id', 
            'attributes_to_store'       => ['id', 'booking'],
            'model' => "App\Models\PayByLink",
            'data_source' => url('/admin/bookings/ajax-bookings-avalon-options'),
            'placeholder' =>'Select booking',
        ]);

```
In the same page I have another select2_json_from_api field but with include_all_form_fields = true

```
$this->crud->addField([
            'label' => 'Pax Name', // Table column heading
            'type' => 'select2_json_from_api',
            'minimum_input_length'    => 0,
            'name' => 'name', // the column that contains the ID of that connected entity
            'attribute'   => "name", // foreign key attribute that is shown to user
            'include_all_form_fields' => true,
            'data_source' => url('/admin/bookings/ajax-bookings-avalon-pax-options'),
        ]);
``
When I click on the second field ('name') this is what I get in my controller from the first field: 
	{"id":"MURR230044149-1","booking":"MURR230044149-1 Localizador: MUR0002615"}


But i f for some reason I reload (for example if  I forgot to fill any  mandatory field and then click  again in name i receive this, note that now I get extra square brakets on the first field booking.

	[{"id":"MURR230044149-1","booking":"MURR230044149-1 Localizador: MUR0002615"}]


### PHP VERSION:
8.2.17

### PHP EXTENSIONS:
Core, date, libxml, openssl, pcre, zlib, filter, hash, json, pcntl, random, Reflection, SPL, session, standard, sodium, mysqlnd, PDO, xml, bcmath, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, iconv, imap, intl, ldap, exif, mysqli, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, readline, shmop, SimpleXML, soap, sockets, sqlite3, sqlsrv, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, pdo_sqlsrv, Zend OPcache

### LARAVEL VERSION:
10.48.14.0

### BACKPACK PACKAGE VERSIONS:
backpack/backupmanager: v5.0.3
backpack/basset: 1.3.4
backpack/calendar-operation: 1.0.5
backpack/crud: 6.7.17
backpack/devtools: 3.1.5
backpack/editable-columns: 3.0.9
backpack/generators: v4.0.5
backpack/logmanager: v5.0.2
backpack/permissionmanager: 7.2.1
backpack/pro: 2.2.4
backpack/settings: 3.1.1
backpack/theme-coreuiv2: 1.2.4
backpack/theme-coreuiv4: 1.1.1
backpack/theme-tabler: 1.2.10

**Comment by karandatwani92** [comment]
Hello @miquelangeld 

Can you rephrase? I'm unable to get the problem.

Can you also share the CRUD code so we can know more about the issue?

**Comment by pxpm** [comment]
ping @karandatwani92

**Comment by pxpm** [comment]
Hey @miquelangeld sorry for the time it took to solve this issue. It seems it felt through the cracks 😢 

I've now released backpack/pro 2.2.29 with the fix for the inconsistent behavior. Thank you very much for the time spent raising and explaining how to reproduce the issue 🙏 

Let us know if you experience some other issues and we hopefully can fix them faster 😃 

Cheers

**Comment by adriallongarriu** [comment]
> Hey @miquelangeld sorry for the time it took to solve this issue. It seems it felt through the cracks 😢
> 
> I've now released backpack/pro 2.2.29 with the fix for the inconsistent behavior. Thank you very much for the time spent raising and explaining how to reproduce the issue 🙏
> 
> Let us know if you experience some other issues and we hopefully can fix them faster 😃
> 
> Cheers

@pxpm there is and error with this change. I was facing the same issue i debuged and found the select is empty(on edit or saving the form without having filled in a required field) because is calling a undeclared variable.

Actual code:
```php
@if ($value)
    @if (is_object($value))
        <option value="{{ json_encode($value) }}" selected>
            {{ $item->{$field['attribute']} ?? '' }}
        </option>
    @else
        @foreach ($value as $item)
            @php
                $item = is_string($item) ? json_decode($item) : (object) $item;
            @endphp
            <option value="{{ json_encode($item) }}" selected>
                {{ $item->{$field['attribute']} ?? '' }}
            </option>
        @endforeach
    @endif
@endif
```
The variable `$item` is declared inisde the loop not before. For now i override the file.
I change the `$item` to use `$value`:
```php
@if (is_object($value))
    <option value="{{ json_encode($value) }}" selected>
        {{ $value->{$field['attribute']} ?? '' }}
    </option>
@else
...
```
